### DISCUSSION TITLE
VPN issue

### DISCUSSION THREAD

**Discussion by shuvashish76** [slowscript/warpinator-android#108] [General]
https://github.com/slowscript/warpinator-android/blob/master/connection-issues.md

> Do not use a VPN (unless you know what you are doing).

Sorry, I use [RethinkDNS](https://github.com/celzero/rethink-app) app which isn't a real vpn, it uses android's VPN to block ads with [hostlists](https://github.com/serverless-dns/blocklists) & also to firewall (_block internet access of_) apps.

These days this is the best solution for non-rooted devices to block ads from all apps. There are many similar opensource apps available e.g. [Blokada](https://github.com/blokadaorg/blokada), [InviZible Pro](https://github.com/Gedsh/InviZible), [NetGuard](https://github.com/M66B/NetGuard), [pdnsf](https://github.com/IngoZenz/personaldnsfilter), [TrackerControl](https://github.com/TrackerControl/tracker-control-android) etc...

![IMG_20220429_043842_296](https://user-images.githubusercontent.com/48226312/165862027-66e8dab9-9fca-45ae-b9a5-dfbc028c9d42.jpg)

All these Firewall rules are added to Gufw Firewall.

When I tried to connect from Android device it gives me this error (_when VPN on_)

![Screenshot_2022-04-29-03-47-36](https://user-images.githubusercontent.com/48226312/165862384-21caa2f4-c5ad-44b6-a8d0-a0ee9b4e33af.jpeg)

![Screenshot_2022-04-29_04-43-13_1](https://user-images.githubusercontent.com/48226312/165862480-ac652076-61f9-4bf1-9777-e7f8fc08bb74.png)

But if I disabe VPN & try to connect then it works & it doesn't stop if I turn on VPN again while both devices are connected.

I know it's a minor thing I could disable VPN service > transfer files > then turn on VPN again but this isn't an solution to the problem & would be somewhat annoying.

Any help/hack for this would be appreciated.
Thanks.

**Comment by slowscript** [comment]
I've installed RethinkDNS as well and can confirm connection not working (although in other ways than you have shown). When I analyzed the traffic with Wireshark (although only briefly), I saw everything I needed, so I'm not sure what the problem is. There however is a solution that worked for me, which was to add Warpinator to Excluded apps list (different than whitelist).

**Reply by ignoramous** [reply]
*rdns dev here*

Rethink doesn't really play nice with LAN or Link Local connections. At least we don't test them yet, but we should.

One option is to expose a setting that lets users exclude local and link-local connections from Rethink: https://github.com/celzero/rethink-app/issues/26#issuecomment-862109640 (this won't work if *Block connections without VPN*, aka Android 9+ VPN *Lockdown* mode, is enabled).

Another option is to test LAN-only apps (like *warpinator*) and see what changes are needed to make the [underlying network engine](https://github.com/celzero/firestack) play ball.

**Comment by shuvashish76** [comment]
Thanks, @slowscript please convert this issue to [discussion](https://github.com/slowscript/warpinator-android/discussions) useful for other users who facing anytype of VPN issue or possible solution discussion.
